tests/context/server.rs

```diff
@@ -0,0 +1,115 @@
+use echo_server::env::Config;
+
+use {
+    anyhow::anyhow,
+    futures_util::TryFutureExt,
+    std::net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4, TcpListener},
+    tokio::{
+        runtime::Handle,
+        sync::broadcast,
+        time::{sleep, Duration},
+    },
+};
+
+pub struct EchoServer {
+    pub public_addr: SocketAddr,
+    shutdown_signal: tokio::sync::broadcast::Sender<()>,
+    is_shutdown: bool,
+}
+
+impl EchoServer {
+    pub async fn start() -> Self {
+        let public_port = get_random_port();
+        let rt = Handle::current();
+        let public_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), public_port);
+
+        let (signal, shutdown) = broadcast::channel(1);
+
+        std::thread::spawn(move || {
+            rt.block_on(async move {
+                let config: Config = Config {
+                    port: public_port,
+                    log_level: "INFO".into(),
+                    relay_url: "https://relay.walletconnect.com".into(),
+                    database_url: "postgres://localhost:5432/postgres".into(),
```

Comment:
```suggestion
                    database_url: "postgres://user:pass@localhost:5432/postgres".into(),
```
This stops the need for the `.username()` as per the comment above